src/System.Private.ServiceModel/src/System/ServiceModel/Security/SecurityProtocolFactory.cs

```diff
@@ -686,6 +693,31 @@ public virtual void OnClose(TimeSpan timeout)
             }
         }
 
+        public virtual Task OnCloseAsync(TimeSpan timeout)
+        {
+            return OnCloseAsyncInternal(timeout);
+        }
+
+        private async Task OnCloseAsyncInternal(TimeSpan timeout)
+        {
+            TimeoutHelper timeoutHelper = new TimeoutHelper(timeout);
+            if (!_actAsInitiator)
+            {
+                foreach (SupportingTokenAuthenticatorSpecification spec in _channelSupportingTokenAuthenticatorSpecification)
+                {
+                    await SecurityUtils.CloseTokenAuthenticatorIfRequiredAsync(spec.TokenAuthenticator, timeoutHelper.RemainingTime());
```

Comment:
I think it's probably worth converting to it eventually, but I'm fine with leaving this as a work item for later given the state of the temporary implementation